packages/providers/src/provider.ts

```diff
@@ -60,6 +86,32 @@ const processGqlReceipt = (gqlReceipt: GqlReceiptFragmentFragment): TransactionR
   }
 };
 
+const processGqlChain = (chain: GqlGetInfoQuery['chain']) => ({
+  name: chain.name,
+  baseChainHeight: BigInt(chain.baseChainHeight),
+  peerCount: chain.peerCount,
+  consensusParameters: {
+    gasPriceFactor: BigInt(chain.consensusParameters.gasPriceFactor),
+    maxGasPerTx: BigInt(chain.consensusParameters.maxGasPerTx),
+    maxScriptLength: BigInt(chain.consensusParameters.maxScriptLength),
+  },
+  latestBlock: {
+    id: chain.latestBlock.id,
+    height: BigInt(chain.latestBlock.height),
+    producer: chain.latestBlock.producer,
+    time: chain.latestBlock.time,
+    transactions: chain.latestBlock.transactions.map((i) => ({
+      id: i.id,
+    })),
+  },
+});
+
+const processNodeInfo = (nodeInfo: GqlGetInfoQuery['nodeInfo']) => ({
+  minBytePrice: BigInt(nodeInfo.minBytePrice),
+  minGasPrice: BigInt(nodeInfo.minGasPrice),
+  nodeVersion: nodeInfo.nodeVersion,
```

Comment:
in this case, will get inherited from the GqlGetInfo would be a string. Do you think is best to create types for these returns? I was wondering if it would be good or not.